### PR TITLE
Move structs from "valuesMulti" and "valuesSingle". Merge input & output

### PR DESCRIPTION
**Pull request by andreibancioiu**
This is a PR from a series of many.

 - InputMultiValue, OutputMultiValue -> MultiValue
 - InputVariadicValue, OutputVariadicValue -> VariadicValue
 - InputOptionalValue, OutputOptionalValue -> OptionalValue

No logic change, only refactoring.

See: https://github.com/multiversx/mx-sdk-abi-go/issues/5.

### CODE REVIEW

File: abi/serializer.go [hunk 1]
```diff
@@ -124,22 +124,23 @@ func (s *serializer) doDeserialize(partsHolder *partsHolder, outputValues []any)
...
-		case *OutputVariadicValues:
+			err = s.deserializeOptionalValue(partsHolder, value)
+		case *MultiValue:
+			err = s.deserializeMultiValue(partsHolder, value)
+		case *VariadicValues:
 			if i != len(outputValues)-1 {
 				return errors.New("variadic values must be last among output values")
 			}
 
+			err = s.deserializeVariadicValues(partsHolder, value)
```
Comment (andreibancioiu) [inline abi/serializer.go:150]: Was dropped by mistake within this series of PRs.

https://github.com/multiversx/mx-sdk-abi-go/blob/main/abi/serializer.go#L147

File: abi/field.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package abi
+
+// Field is a field in a struct, enum etc.
+type Field struct {
```
Comment (andreibancioiu) [inline abi/field.go:4]: Was moved from `valuesSingle` to its own file.

Most other structs from `valuesSingle` were moved to their own files (`addressValue`, `boolValue` etc.) in previous  PRs. Small integers follow in the next PR.

**Review by sstanculeanu — DISMISSED** [review]

File: abi/serializer.go [hunk 3]
```diff
@@ -154,15 +155,15 @@ func (s *serializer) doDeserialize(partsHolder *partsHolder, outputValues []any)
...
 		return nil
 	}
 
 	return s.doSerialize(partsHolder, []any{value.Value})
 }
 
-func (s *serializer) serializeInputMultiValue(partsHolder *partsHolder, value InputMultiValue) error {
+func (s *serializer) serializeMultiValue(partsHolder *partsHolder, value *MultiValue) error {
 	for _, item := range value.Items {
 		err := s.doSerialize(partsHolder, []any{item})
```
Comment (axenteoctavian) [inline abi/serializer.go]: Can't you do `return s.doSerialize(partsHolder, value.Items)` maybe? Same comment for the other funcs

Comment (andreibancioiu) [reply abi/serializer.go]: Very good point! Refactored.

Tests are passing in the "staging" branch: https://github.com/multiversx/mx-sdk-abi-go/pull/8.

**Review by sstanculeanu — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**